src/main/java/org/mule/extension/socket/api/SocketOperations.java

```diff
@@ -33,19 +33,14 @@
    * operation will return a {@code null} payload.
    *
    * @param content        data that will be serialized and sent through the socket.
-   * @param outputEncoding encoding that will be used to serialize the {@code data} if its type is {@link String}.
    * @throws ConnectionException if the connection couldn't be established, if the remote host was unavailable.
    */
+  @DataTypeParameters
   public Result<InputStream, SocketAttributes> sendAndReceive(@Connection RequesterConnection connection,
-                                                              @Config RequesterConfig config,
-                                                              @Content Object content,
-                                                              @Optional @Summary("Encoding to use when the data to serialize is of String type") String outputEncoding)
+                                                              @Content InputStream content)
       throws ConnectionException, IOException {
     SocketClient client = connection.getClient();
-
-    outputEncoding = override(outputEncoding, config.getDefaultEncoding());
-
-    client.write(content, outputEncoding);
+    client.write(content);
```

Comment:
the connector is receiving a plain InputStream and send the byte of it, so encoding doesn't matter